### ISSUE TITLE
New or reconnected tabs are always placed at the end of the tab strip

### ISSUE THREAD

**Issue by C4rb0n** [mRemoteNG/mRemoteNG#2159]
New or reconnected tabs are placed at the end of the tab strip of the associated panel, regardless of configured preference.  This is a regression introduced in published nightly v1.77.2-nb.

## Expected Behavior

1. Connections should be open to the right the active tab when the "Open new tab to the right of the currently selected tab" preference is enabled.
2. "Reconnect" should do so with the tab in its current position.

## Current Behavior

1. New tabs are placed at the far right end of the tab strip.
2. "Reconnect" moves the reconnected tab to the far right end of the tab strip.

## Possible Solution

This is a regression from behavior up to and including v1.77.1.

## Steps to Reproduce (for bugs)

1. Make sure the "Open new tab to the right of the currently selected tab" option is enabled
2. Open five different connections in the same panel
3. Select the left-most tab
4. Open another connection from the connection list
5. Notice that the tab ends up at the far right
6. Select the left-most tab
7. Right-click the tab and select "Reconnect"
8. Notice that the tab ends up at the far right.

## Context

Breaks workflow, causes confusion, requires hunting around for tabs on a busy tab strip.

## Your Environment

* Version used:
* Operating System and version (e.g. Windows 10 1709 x64):
MRemoteNG Nightly v1.77.2-nb
Windows 10 Pro Version 21H2 (OS Build 19044.1348)

**Comment by simonai1254** [comment]
Noticed this as well (and thought I stumbled across this in another issue though couldn't find it anymore), though I was aware of it after a short time and don't have that many tabs open (or enough screen real estate) that it didn't bother me too much. Independently I would support it to get this resolved.

**Comment by nelgin** [comment]
I am going to +1 to get this fixed as soon as possible. It's very annoying behavior. I open a lot of tabs and in a specific order so I know which is which easily, I can have 12 or more tabs open. If I get disconnected from any one of those servers and I reconnect, the tab is placed at the end and I have to move it back. For a server with 5 connections, this disrupts my workflow.

**Comment by dawk0** [comment]
Great, I found someone you've already created an issue for that.
I am an Application Manager working with a lot of servers simultaneously  - that behaviour absolutely confuses me (+ those when I share my screen).
Please fix it, or give me a setting to deactivate that behaviour.

**Comment by Kvarkas** [comment]
have you check that setting in options?
![image](https://user-images.githubusercontent.com/3611964/163409553-ec789f60-10d1-4a57-812e-d9317b5e5c98.png)

**Comment by dawk0** [comment]
> have you check that setting in options? ![image](https://user-images.githubusercontent.com/3611964/163409553-ec789f60-10d1-4a57-812e-d9317b5e5c98.png)

Yes, unfortunately it does not do the trick. When turning that setting off and reconnecting to an existing tab the position still jumps to the far right.

**Comment by Kvarkas** [comment]
ok will double check that for nb

**Comment by nelgin** [comment]
Yes, I've tried a number of settings and it's still opening the tabs to the right. I just wonder why this has been waiting for so long for someone to look into. It's really annoying.

**Comment by nelgin** [comment]
@Kvarkas Checking my config, yes. I have that option checked here.

**Comment by nelgin** [comment]
This has been lingering for almost 4 months now. It's really hindering my workflow. I guess it's fine to find an alternative that actually works.

**Comment by nelgin** [comment]
New build...and still not fixed. Yeah, I'm done. Over and out.

**Comment by dawk0** [comment]
> New build...and still not fixed. Yeah, I'm done. Over and out.

Yeah, I'm pretty disappointed too

**Comment by nelgin** [comment]
The only problem, I'm finding it difficult to find a suitable replacement. I'm trying Tabby right now, but you cannot configure per-session fonts by the looks of it.

**Comment by simonai1254** [comment]
> This has been lingering for almost 4 months now. It's really hindering my workflow. I guess it's fine to find an alternative that actually works.

Please just consider that this is a project maintained by a single person in it's spare time, and (unfortunately) there are more severe bugs than just an annoying wrong tab order. You may contribute a little more motivation with a donation or fix the code itself, but just because it's free and does not (yet) have all features working correctly that a commercial solution has in my point of view is not a reason to completely ditch the utility...

**Comment by nelgin** [comment]
Or I can just go find something that works.  Heading towards 5 months now.

**Comment by Kvarkas** [comment]
well until waiting lazy dev's you may learn how to program and fix that annoying issue and submit PR that community will say a big thanks for you, if you are lazy for such then please wait till we fix bigger issues first, then move to this one, thanks :)

**Comment by nelgin** [comment]
Or I can just use something else. I think this conversation is over.

**Comment by Kvarkas** [comment]
well, I may understand if you paid for solution but you are using it for free, but, we are very democratic community, promise not pursuing you if you find something else :D

**Comment by robertpopa22** [comment]
**Fix available for testing**

A fix for this issue has been implemented in our community fork (`robertpopa22/mRemoteNG`, branch `main`). A beta build including this fix is available from the fork's Releases page.

Please test and report if this resolves your issue.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*
